esmvaltool/recipes/recipe_climwip_brunner20esd.yml

```diff
@@ -0,0 +1,202 @@
+
+preprocessors:
+  climwip_general: &general
+    regrid:
+      target_grid: 2.5x2.5
+      scheme: linear
+
+  climatological_mean:
+    <<: *general
+    climate_statistics:
+      operator: mean
+
+  detrended_std:
+    custom_order: true
+    <<: *general
+    annual_statistics:
+      operator: mean
+    detrend:
+      dimension: time
+      method: linear
+    climate_statistics:
+      operator: std_dev
+
+  annual_trend:
+    <<: *general
+    annual_statistics:
+      operator: mean
+    linear_trend:
+      coordinate: time
+    convert_units:
+      units: "K year-1"
+
+  global_mean:
+    <<: *general
+    climate_statistics:
+      operator: mean
+    area_statistics:
+      operator: mean
+
+  temperature_anomalies:
+    custom_order: true
+    area_statistics:
+      operator: mean
+    annual_statistics:
+      operator: mean
+    anomalies:
+      period: full
+      reference:
+        start_year: 1995
+        start_month: 1
+        start_day: 1
+        end_year: 2014
+        end_month: 12
+        end_day: 31
+      standardize: false
+
+diagnostics:
+  # if local anomalies are needed as predictors they need to be calculated first
+  calculate_local_anomalies_tas:
+    variables:
+      tas_CLIM: &performance_settings
+        short_name: tas
+        start_year: 1995
+        end_year: 2014
+        mip: Amon
+        preprocessor: climatological_mean
+        additional_datasets: *obs_data
+      tas_GLOBAL:
+        <<: *performance_settings
+        preprocessor: global_mean
+    scripts:
+      local_anomalies:
+        script: weighting/calculate_difference_variable_group.py
+        obs_data: native6
+
+  calculate_local_anomalies_psl:
+    variables:
+      psl_CLIM:
+        <<: *performance_settings
+        short_name: psl
+        preprocessor: climatological_mean
+      psl_GLOBAL:
+        <<: *performance_settings
+        short_name: psl
+        preprocessor: global_mean
+    scripts:
+      local_anomalies:
+        script: weighting/calculate_difference_variable_group.py
+        obs_data: native6
+
+  calculate_weights_climwip:
+    variables:
+      tas_CLIM_i: &independence_settings
+        short_name: tas
+        start_year: 1995
+        end_year: 2014
+        mip: Amon
+        preprocessor: climatological_mean
+        additional_datasets: *obs_data
+      psl_CLIM_i:
+        <<: *independence_settings
+        short_name: psl
+      tas_STD:
+        <<: *performance_settings
+        preprocessor: detrended_std
+      psl_STD:
+        <<: *performance_settings
+        short_name: psl
+        preprocessor: detrended_std
+      tas_TREND:
+        <<: *performance_settings
+        preprocessor: annual_trend
+
+    scripts:
+      climwip:
+        script: weighting/climwip/main.py
+        obs_data: native6
+#        ancestors: [calculate_local_anomalies_tas/tas_CLIM,
+#                    calculate_local_anomalies_psl/psl_CLIM,
```

Comment:
I think this will return the variable groups as they are *before* entering the diagnostic script. I guess you want `calculate_local_anomalies/local_anomalies` as ancestor, which is the task for running the diagnostic script.